### PR TITLE
[interp] fix i4<>i8 conversion for builtin types, it was inverted

### PR DESCRIPTION
**Pull request by lewurm**
needs backport to `2018-02` (and `2018-04`?)

### CODE REVIEW

File: mono/mini/interp/transform.c [hunk 1]
```diff
@@ -943,7 +943,7 @@ interp_transform_call (TransformData *td, MonoMethod *method, MonoMethod *target
 				if (arg_size > SIZEOF_VOID_P) { // 8 -> 4
 					switch (type_index) {
 					case 0: case 1:
-						ADD_CODE (td, MINT_CONV_I8_I4);
+						ADD_CODE (td, MINT_CONV_I4_I8);
 						break;
```
Comment (vargaz) [inline mono/mini/interp/transform.c]: The opcodes could use a better name like MINT_CONV_X_TO_Y.

**Comment by luhenry** [comment]
@monojenkins build OS X i386

**Comment by lewurm** [comment]
@monojenkins backport `2018-02`

**Comment by lewurm** [comment]
@monojenkins backport `2018-04`

**Comment by lewurm** [comment]
@vargaz I'm gonna change that on `master`

**Comment by lewurm** [comment]
@monojenkins backport 2018-02

**Comment by lewurm** [comment]
@monojenkins backport 2018-04

**Pull request merged**